wrapper.ts

```diff
@@ -70,27 +73,27 @@ export class Pty {
     // Transfer ownership of the FD to us.
     this.#fd = this.#pty.takeFd();
 
-    this.#socket = new ReadStream(this.#fd);
-    const userFacingRead = new PassThrough();
-    const userFacingWrite = new PassThrough();
-    this.#socket.pipe(userFacingRead);
-    userFacingWrite.pipe(this.#socket);
-    this.read = userFacingRead;
-    this.write = userFacingWrite;
+    this.read = new ReadStream(this.#fd);
+    this.write = new WriteStream(this.#fd);
```

Comment:
warning: the pty implementation handles muxing both internally. please make sure this is safe and won't lead to the FD being closed twice!